src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationOptions.cs

```diff
@@ -152,5 +153,13 @@ public string CookieName
         /// to the client. This can be used to mitigate potential problems with very large identities.
         /// </summary>
         public IAuthenticationSessionStore SessionStore { get; set; }
+
+        CookieAuthenticationOptions IOptions<CookieAuthenticationOptions>.Value
```

Comment:
> Oh, well we already have precedence for this weird options implementing IOptions of itself already, so that's why I just went with that...

Having to implement `IOptions<>` in every options class is a real PITA and not really DRY IMHO, since you end up duplicating the same code again and again (even in the Caching repo!).

> I'm not sure its worth another wrapper, ConfigureOptions is already a plain wrapper now :/

You could see it as `IOptions<>`'s default implementation :smile: